### PR TITLE
Readthedocs

### LINKED ISSUE #155: Documentation
This seriously needs documentation, like any little bit:
- [ ] api
- [ ] mwe
- [ ] advanced features

For now just some simple markdown files in a `/docs` folder will be appreciated. I can contribute to the CI and sphinx formatting.

### PR DESCRIPTION
**Pull request by LecrisUT**
Hi, this is a very rudimentary migration of the wiki pages to within source

It should be compatible with readthedocs and once it is enabled, we can discuss any style changes that are wanted.

Closes #155 

TODO:
- [x] Import/Export man page to and from markdown format (`pandoc`)

### CODE REVIEW

**Review by sopos — COMMENTED** [review]
I suspect this commit is not really related to the readthedocs stuff.

**Comment by LecrisUT** [comment]
It is related. That one creates the python optional dependencies so that rtd can pull in the correct verisions of sphinx, etc. in the [`.readthedocs.yaml`](https://github.com/beakerlib/beakerlib/pull/157/commits/a624315142f9e902e5ec07c601a3b9704d3c95ae#diff-03efc769b870804394632e45d7885272b44c16939517fb31c9d7c614d2ffae57R5-R7) file

**Comment by sopos** [comment]
I have no clue how this works but it would be nice to generate the documentation from the sources directly.
The content of _man.pod_ is actually the _src/docs/pod/beakerlib.pod_ which is built by `make`.

**Comment by LecrisUT** [comment]
> The content of _man.pod_ is actually the _src/docs/pod/beakerlib.pod_ which is built by `make`.

Thanks for the tip. Can you expand on this a bit, how is `beakerlib.pod` generated?  Can it output a markdown file format? And what programs are needed to run relevant part of the `make` script?

> it would be nice to generate the documentation from the sources directly

About that, could you enable the integration on https://readthedocs.org/ with PR support?

**Comment by sopos** [comment]
> > The content of _man.pod_ is actually the _src/docs/pod/beakerlib.pod_ which is built by `make`.
> 
> Thanks for the tip. Can you expand on this a bit, how is `beakerlib.pod` generated? Can it output a markdown file format? And what programs are needed to run relevant part of the `make` script?

Following will create the pod file. POD is the inline documentation which may be converted to various format using `pod2*` tools.
```
cd src
make docs/pod/beakerlib.pod
```
AFAIK, `perl` is needed for it to work.

> 
> > it would be nice to generate the documentation from the sources directly
> 
> About that, could you enable the integration on https://readthedocs.org/ with PR support?

I created a project there and enabled the PR builds. Otherwise I have no experience with that. Not sure what all need to be set there. I may give you rights there.

**Comment by LecrisUT** [comment]
https://beakerlib--157.org.readthedocs.build/en/157/index.html

There we go. I'll try to get the pod2pandoc working tomorrow. Now for the actual documentation page, any preference or reference you want to have implemented?

**Comment by sopos** [comment]
> https://beakerlib--157.org.readthedocs.build/en/157/index.html
> 
> There we go. I'll try to get the pod2pandoc working tomorrow. Now for the actual documentation page, any preference or reference you want to have implemented?

What kind of reference do you mean? Personally I have no preferences. As this is completely new thing, I would leave it on you. We can discuss / tweak the result later, if necessary.

**Comment by LecrisUT** [comment]
> What kind of reference do you mean?

I was thinking if you had a documentation page you like and want to emulate.

Well the framework is more-or-less in place. Now it's just to actually make the pages as we want to. I would have preferred to use pandoc to get more control over the format, but `pod2pandoc` is not available on fedora for me to try it out.

Anyway, review please?

**Comment by sopos** [comment]
Seems fine to me.  @hegerj, could you take a look as well?

File: doc/index.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+maxdepth: 2
+titlesonly: true
+caption: Contents
+glob: true
+---
+manual
+roadmap
+:::
+
+## News
```
Comment (sopos) [inline doc/index.md:20]: I'm just wondering if this should not be moved / linked to elsewhere. Normally I create the release and the just refer to it so I do not need to touch the repo itself again.

Comment (LecrisUT) [reply doc/index.md:20]: Sure, we can move it as needed. Are these created from the spec file changelog? Is there a way of extracting them and/or making them in a markdown format? I know packit can do the opposite.

File: doc/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/man.md
```
Comment (sopos) [inline doc/.gitignore:1]: apparently there's a missing line break which is a good practice to have

File: doc/_static/.gitkeep [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (sopos) [inline doc/_static/.gitkeep:1]: do we really need this empty file?

File: doc/index.md (hunk 1 above)
Comment (LecrisUT) [reply doc/index.md:20]: > No, it is a manual work.

Let me rephrase that. Is the intention to have it synchronized, or is there cases where one is more detailed than the other? Is one a subset of the other? E.g. I can implement a roadmap section, a news section and a changelog section, where only one of them is synchronized.

As for the format. Which one do you want as different pages?

File: doc/.gitignore (hunk 2 above)
Comment (LecrisUT) [reply doc/.gitignore:1]: Good catch, I usually rely on `pre-commit` to catch these things, but I forgot this one does not have it implemented yet.

File: doc/_static/.gitkeep (hunk 3 above)
Comment (LecrisUT) [reply doc/_static/.gitkeep:1]: Yes, the folder needs to exist for `sphinx-build` to proceed. But it is temporary, because probably there will be a logo or something that will be added there soon.

File: doc/index.md (hunk 1 above)
Comment (sopos) [reply doc/index.md:20]: Hm, I'll try this way to maintain it. I may be better, after all. We can change it anytime in the future.

Comment (LecrisUT) [reply doc/index.md:20]: Ok, np. What about

> As for the format. Which one do you want as different pages?

Comment (sopos) [reply doc/index.md:20]: I do not have any preference, I think the MD is fine. Just KISS works.

It could be populated as a part of the individual changes in some "upcoming" section and finalized with just before the actual release.

File: doc/_static/.gitkeep (hunk 3 above)
Comment (sopos) [reply doc/_static/.gitkeep:1]: Hmm, logo. For that we would need someone with design skills.

Comment (LecrisUT) [reply doc/_static/.gitkeep:1]: Artists have to survive on something right? :D

But there are things that should be added to at least silent some browsers, like `favicon`, `robots.txt`, etc.

File: doc/index.md (hunk 1 above)
Comment (LecrisUT) [reply doc/index.md:20]: Ok, I'll just quickly fix the `.gitignore`. Do feel free to request any changes, no matter how big or nitpicky they are. It is very likely that it is something that I could use for my own projects, and I tend to be more productive when I work on other people's stuff.

File: doc/_static/.gitkeep (hunk 3 above)
Comment (sopos) [reply doc/_static/.gitkeep:1]: I wonder if there is something like ChatGPT which could draw images based on a text description. :-)

Comment (LecrisUT) [reply doc/_static/.gitkeep:1]: I mean there are dally and other plenty of "prompt" art. But I don't even trust my artistic eye to identify a child's drawing from a Picasso, so I will deffer on that :D. Still I would rather have something commissioned than research the legalities.

**Comment by hegerj** [comment]
Looks good to me, however I too have limited understanding of how it works.

**Comment by sopos** [comment]
@psss can I ask you for a quick review as someone with the _readthedocs_ knowledge?

**Comment by LecrisUT** [comment]
@psss when you have the time, I had the idea for https://github.com/beakerlib/libraries/issues/13, since it can touch tmt stuff as well. Would be nice to make them more visible and get more people on board to have reusable tests there.

**Comment by psss** [comment]
Yeah, good idea to make them more visible. Added a comment there.

File: doc/.gitignore (hunk 2 above)
Comment (LecrisUT) [reply doc/.gitignore:1]: Oh, thanks for reminding, I completely forgot :D

Comment (LecrisUT) [reply doc/.gitignore]: Ok, that's fixed now. Testing-farm is failing, but that's another can of worms to address in #160
